Utils/github_workflow_scripts/utils.py

```diff
@@ -120,19 +120,21 @@ def __init__(self, repo: Repo, branch_to_checkout: str, fork_owner: str | None =
                 self.repo.create_remote(name=forked_remote_name, url=url)
                 print(f'Successfully created remote {forked_remote_name} for repo {url}')  # noqa: T201
             except Exception as error:
-                print(f'could not create remote from {url}, {error=}')  # noqa: T201
-                # handle the case where the name of the forked repo is not content
-                if github_event_path := os.getenv("GITHUB_EVENT_PATH"):
-                    try:
-                        payload = json.loads(github_event_path)
-                    except ValueError:
-                        print('failed to load GITHUB_EVENT_PATH')  # noqa: T201
-                        raise ValueError(f'cannot checkout to the forked branch {branch_to_checkout} of the owner {fork_owner}')
-                    # forked repo name includes fork_owner + repo name, for example foo/content.
-                    forked_repo_name = payload.get("pull_request", {}).get("head", {}).get("repo", {}).get("full_name")
-                    self.repo.create_remote(name=forked_remote_name, url=f"https://github.com/{forked_repo_name}")
-                else:
-                    raise
+                if f'{forked_remote_name} already exists' not in str(error):
```

Comment:
can you explain this if?

Reply:
Yes,
when we call the checkout - it first creates a new forked_remote_branch name in line 117:
forked_remote_name = f'{fork_owner}_{repo_name}_{branch_to_checkout}_remote'

so if there was already a try for checkout, for example when checking the support level, the next try to checkout will fail and return an error.

In this case the except clause will ignore the "already exists" error and would continue to line 139 to checkout.

Reply:
add this as a comment to the code